receiver/hostmetricsreceiver/internal/scraper/virtualmemoryscraper/virtualmemory_scraper_windows.go

```diff
@@ -0,0 +1,190 @@
+	s.prevPagingScrapeTime = time.Now()
+
+	var err error
+
+	s.pageReadsPerSecCounter, err = pdh.NewPerfCounter(pageReadsPerSecPath, true)
+	if err != nil {
+		return err
+	}
+
+	s.pageWritesPerSecCounter, err = pdh.NewPerfCounter(pageWritesperSecPath, true)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Close
+func (s *scraper) Close(_ context.Context) error {
+	var errors []error
+
+	err := s.pageReadsPerSecCounter.Close()
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	err = s.pageWritesPerSecCounter.Close()
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	return componenterror.CombineErrors(errors)
+}
+
+// ScrapeMetrics
+func (s *scraper) ScrapeMetrics(ctx context.Context) (pdata.MetricSlice, error) {
+	_, span := trace.StartSpan(ctx, "virtualmemoryscraper.ScrapeMetrics")
+	defer span.End()
+
+	metrics := pdata.NewMetricSlice()
+
+	var errors []error
+
+	err := s.scrapeAndAppendSwapUsageMetric(metrics)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	err = s.scrapeAndAppendPagingMetric(metrics)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	if len(errors) > 0 {
+		return metrics, componenterror.CombineErrors(errors)
+	}
+
+	return metrics, nil
+}
+
+func (s *scraper) scrapeAndAppendSwapUsageMetric(metrics pdata.MetricSlice) error {
+	pageFiles, err := getPageFileStats()
+	if err != nil {
+		return err
+	}
+
+	idx := metrics.Len()
+	metrics.Resize(idx + 1)
+	initializeSwapUsageMetric(metrics.At(idx), pageFiles)
+	return nil
+}
+
+func initializeSwapUsageMetric(metric pdata.Metric, pageFiles []*pageFileData) {
+	metricSwapUsageDescriptor.CopyTo(metric.MetricDescriptor())
+
+	idps := metric.Int64DataPoints()
+	idps.Resize(2 * len(pageFiles))
+
+	idx := 0
+	for _, pageFile := range pageFiles {
+		initializeSwapUsageDataPoint(idps.At(idx+0), pageFile.name, usedLabelValue, int64(pageFile.used))
+		initializeSwapUsageDataPoint(idps.At(idx+1), pageFile.name, freeLabelValue, int64(pageFile.total-pageFile.used))
+		idx += 2
+	}
+}
+
+func initializeSwapUsageDataPoint(dataPoint pdata.Int64DataPoint, pageFileLabel string, stateLabel string, value int64) {
+	labelsMap := dataPoint.LabelsMap()
+	labelsMap.Insert(pageFileLabelName, pageFileLabel)
+	labelsMap.Insert(stateLabelName, stateLabel)
+	dataPoint.SetTimestamp(pdata.TimestampUnixNano(uint64(time.Now().UnixNano())))
+	dataPoint.SetValue(value)
+}
+
+func (s *scraper) scrapeAndAppendPagingMetric(metrics pdata.MetricSlice) error {
+	now := time.Now()
+	durationSinceLastScraped := now.Sub(s.prevPagingScrapeTime).Seconds()
+	s.prevPagingScrapeTime = now
+
+	pageReadsPerSecValues, err := s.pageReadsPerSecCounter.ScrapeData()
+	if err != nil {
+		return err
+	}
+
+	pageWritesPerSecValues, err := s.pageWritesPerSecCounter.ScrapeData()
+	if err != nil {
+		return err
+	}
+
+	s.cumulativePageReads += (pageReadsPerSecValues[0].Value * durationSinceLastScraped)
+	s.cumulativePageWrites += (pageWritesPerSecValues[0].Value * durationSinceLastScraped)
```

Comment:
One issue with doing this calculation is that we are now reporting float values which looks & feels a little strange (see screenshot in PR description). Do you think it would be better to convert to an integer here?

Reply:
With the underlying precision being at 1s anyway I don't think you'd lose much to convert them to integers.